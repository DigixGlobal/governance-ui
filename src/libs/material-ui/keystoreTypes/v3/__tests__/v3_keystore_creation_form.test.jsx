import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import V3KeystoreCreationForm from '../v3_keystore_creation_form';

describe('<V3KeystoreCreationForm />', () => {
  test('renders correctly', () => {
    const component = shallow(
      <V3KeystoreCreationForm
        formData={{}}
        formChange={jest.fn}
      />,
    );

    expect(toJson(component)).toMatchSnapshot();
  });
});
