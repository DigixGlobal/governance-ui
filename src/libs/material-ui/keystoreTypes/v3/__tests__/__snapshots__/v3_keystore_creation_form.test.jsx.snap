// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<V3KeystoreCreationForm /> renders correctly 1`] = `
<div>
  <FormField
    formChange={[Function]}
    formData={Object {}}
    label="Name"
    name="name"
    placeholder="Nickname for the Address"
  />
  <FormGroup
    widths="equal"
  >
    <FormField
      formChange={[Function]}
      formData={Object {}}
      label="Password"
      name="password"
      placeholder="Enter Password"
      type="password"
    />
    <FormField
      formChange={[Function]}
      formData={Object {}}
      label="Confirm Password"
      name="confirmPassword"
      placeholder="Confirm Password"
      type="password"
    />
  </FormGroup>
  <Advanced
    title="Advanced"
  >
    <Connect(NetworkTokensSelector)
      formChange={[Function]}
      formData={Object {}}
    />
    <p>
      Optionally override the default crypto and encryption values (do not prefix hex strings):
    </p>
    <FormGroup
      widths="equal"
    >
      <EntropyField
        formChange={[Function]}
        formData={Object {}}
        label="Private Key (hex)"
        name="privateKey"
      />
      <EntropyField
        bits={128}
        formChange={[Function]}
        formData={Object {}}
        label="Initialization Vector (hex)"
        name="iv"
      />
      <EntropyField
        formChange={[Function]}
        formData={Object {}}
        label="Salt (hex)"
        name="salt"
      />
      <FormField
        formChange={[Function]}
        formData={Object {}}
        label="Iteration Count"
        name="n"
        placeholder="1024"
        type="number"
      />
    </FormGroup>
  </Advanced>
</div>
`;
